### DISCUSSION TITLE
Sizing calculator?  Postgres storage size for 1M files ?

### DISCUSSION THREAD

**Discussion by netrc** [juicedata/juicefs#3218] [Q&A] [answered]
Anyone able to provide some simple calculations to size a Postgres cluster for 1 million files, say 100,000 directories?

Actual compute cycles won't be too high - normally low create/delete operations; just occasional/short bursts of high activity.

What else is needed for back-of-envelope calculation?

**Comment by davies** [comment, marked as answer]
It's roughly 500 bytes per file/directory, so 500MB per 1 million files.
